src/pybind/mgr/dashboard/controllers/auth.py

```diff
@@ -28,29 +28,47 @@ class Auth(RESTController):
     """
     Provide authenticates and returns JWT token.
     """
-
     def create(self, username, password):
         user_data = AuthManager.authenticate(username, password)
         user_perms, pwd_expiration_date, pwd_update_required = None, None, None
-        if user_data:
-            user_perms = user_data.get('permissions')
-            pwd_expiration_date = user_data.get('pwdExpirationDate', None)
-            pwd_update_required = user_data.get('pwdUpdateRequired', False)
-
-        if user_perms is not None:
-            logger.debug('Login successful')
-            token = JwtManager.gen_token(username)
-            token = token.decode('utf-8')
-            cherrypy.response.headers['Authorization'] = "Bearer: {}".format(token)
-            return {
-                'token': token,
-                'username': username,
-                'permissions': user_perms,
-                'pwdExpirationDate': pwd_expiration_date,
-                'sso': mgr.SSO_DB.protocol == 'saml2',
-                'pwdUpdateRequired': pwd_update_required
-            }
+        attempt = mgr.ACCESS_CTRL_DB.get_attempt(username)
+        if attempt < 10:
+            if user_data:
+                user_perms = user_data.get('permissions')
+                pwd_expiration_date = user_data.get('pwdExpirationDate', None)
+                pwd_update_required = user_data.get('pwdUpdateRequired', False)
 
+            if user_perms is not None:
+                logger.debug('Login successful')
+                attempt = 0
+                mgr.ACCESS_CTRL_DB.save_attempt(username, attempt)
+                mgr.ACCESS_CTRL_DB.save()
+                token = JwtManager.gen_token(username)
+                token = token.decode('utf-8')
+                cherrypy.response.headers['Authorization'] = "Bearer: {}".format(token)
+                return {
+                    'token': token,
+                    'username': username,
+                    'permissions': user_perms,
+                    'pwdExpirationDate': pwd_expiration_date,
+                    'sso': mgr.SSO_DB.protocol == 'saml2',
+                    'pwdUpdateRequired': pwd_update_required
+                }
+            attempt += 1
+            mgr.ACCESS_CTRL_DB.save_attempt(username, attempt)
+            mgr.ACCESS_CTRL_DB.save()
+        else:
+            try:
+                user = mgr.ACCESS_CTRL_DB.get_user(username)
+                user.enabled = False
```

Comment:
If we reset the attempt to 0 here, we don't need a CLI command later. And the user re-enable workflow can be deal with both from CLI and UI (UI already displays a enabled checkbox). The only thing is to ensure that disabled users get the same error on log-in attempt than regular users with a failed password.

BTW, I see the `.enabled` is set here but not saved anywhere? Isn't it required?

Reply:
Fixed this. Thanks.